x/marketmap/types/msg.go

```diff
@@ -0,0 +1,63 @@
+package types
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+var _ sdk.Msg = &MsgUpdateMarketMap{}
+
+// GetSigners gets the address that must sign this message.
+func (m *MsgUpdateMarketMap) GetSigners() []sdk.AccAddress {
+	// convert from string to acc address
+	addr, _ := sdk.AccAddressFromBech32(m.Signer)
+	return []sdk.AccAddress{addr}
+}
+
+// ValidateBasic determines whether the information in the message is formatted correctly, specifically
+// whether the signer is a valid acc-address.
+func (m *MsgUpdateMarketMap) ValidateBasic() error {
+	// validate signer address
+	if _, err := sdk.AccAddressFromBech32(m.Signer); err != nil {
+		return err
+	}
+
+	for _, market := range m.CreateMarkets {
+		if err := market.Ticker.ValidateBasic(); err != nil {
+			return err
+		}
+
+		if len(market.Paths.Paths) == 0 {
+			return fmt.Errorf("at least one path is required for a ticker to be calculated")
+		}
+
+		for _, path := range market.Paths.Paths {
+			if err := path.ValidateBasic(); err != nil {
+				return err
+			}
+		}
+
+		if uint64(len(market.Providers.Providers)) < market.Ticker.MinProviderCount {
+			return fmt.Errorf("this ticker must have at least %d providers; got %d",
+				market.Ticker.MinProviderCount,
+				len(market.Providers.Providers),
+			)
+		}
+
+		seenProviders := make(map[string]struct{})
+		for _, provider := range market.Providers.Providers {
+			// check for duplicate providers
+			if _, seen := seenProviders[provider.Name]; seen {
+				return fmt.Errorf("duplicate provider found: %s", provider.Name)
+			}
+			seenProviders[provider.Name] = struct{}{}
+
+			if provider.OffChainTicker == "" {
+				return fmt.Errorf("got empty off chain ticker for provider %s", provider.Name)
+			}
```

Comment:
nit: I would just call validate basic on the `ProviderConfig` while doing the duplicate name check